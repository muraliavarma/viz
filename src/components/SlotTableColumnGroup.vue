<script>
export default {
  name: 'SlotTableColumnGroup',
  functional: true,
  props: {
    sticky: {
      type: String,
      default: '',
    },
  },
  render(createElement, ctx) {
    const { sticky } = ctx.props;
    const data = Object.assign({}, ctx.data);
    if (sticky === 'left') {
      data.class = Object.assign({}, data.class, { 'sticky-left': true });
    } else if (sticky === 'right') {
      data.class = Object.assign({}, data.class, { 'sticky-right': true });
    }
    return createElement('th', data, ctx.children);
  },
};
</script>
