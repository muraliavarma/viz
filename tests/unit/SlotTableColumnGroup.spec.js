import { mount } from '@vue/test-utils';
import SlotTableColumnGroup from '@/components/SlotTableColumnGroup.vue';

const click = jest.fn();
const wrapper = mount(SlotTableColumnGroup, {
  context: {
    children: ['foo'],
    on: { click },
    staticClass: 'bar',
    attrs: {
      baz: 'quux',
    },
    props: {
      span: 3,
    },
  },
});

describe('SlotTableColumnGroup.vue', () => {
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('triggers click event', () => {
    wrapper.trigger('click');
    expect(click).toHaveBeenCalled();
  });

  it('does left sticky', () => {
    const stickyWrapper = mount(SlotTableColumnGroup, {
      context: {
        staticClass: 'foo',
        class: { bar: true },
        props: {
          sticky: 'left',
        },
      },
    });
    expect(stickyWrapper.classes('foo')).toBe(true);
    expect(stickyWrapper.classes('bar')).toBe(true);
    expect(stickyWrapper.classes('sticky-left')).toBe(true);
    expect(stickyWrapper.classes('sticky-right')).toBe(false);
  });

  it('does right sticky', () => {
    const stickyWrapper = mount(SlotTableColumnGroup, {
      context: {
        staticClass: 'foo',
        class: { bar: true },
        props: {
          sticky: 'right',
        },
      },
    });
    expect(stickyWrapper.classes('foo')).toBe(true);
    expect(stickyWrapper.classes('bar')).toBe(true);
    expect(stickyWrapper.classes('sticky-right')).toBe(true);
    expect(stickyWrapper.classes('sticky-left')).toBe(false);
  });

  it('does no sticky', () => {
    const stickyWrapper = mount(SlotTableColumnGroup, {
      context: {
        staticClass: 'foo',
        class: { bar: true },
        props: {
        },
      },
    });
    expect(stickyWrapper.classes('foo')).toBe(true);
    expect(stickyWrapper.classes('bar')).toBe(true);
    expect(stickyWrapper.classes('sticky-right')).toBe(false);
    expect(stickyWrapper.classes('sticky-left')).toBe(false);
  });
});
