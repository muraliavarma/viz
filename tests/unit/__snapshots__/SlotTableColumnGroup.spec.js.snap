// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotTableColumnGroup.vue has the expected html structure 1`] = `
<th
  baz="quux"
  class="bar"
>
  foo
</th>
`;
